pkg/controller/servicebindingrequest/reconciler.go

```diff
@@ -0,0 +1,527 @@
+package servicebindingrequest
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"strings"
+
+	"github.com/go-logr/logr"
+	osappsv1 "github.com/openshift/api/apps/v1"
+	olmv1alpha1 "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	extv1beta1 "k8s.io/api/extensions/v1beta1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+
+	v1alpha1 "github.com/redhat-developer/service-binding-operator/pkg/apis/apps/v1alpha1"
+)
+
+const (
+	connectsToLabel                = "connects-to"
```

Comment:
It is currently hard coded right?

Reply:
@Avni-Sharma The intention is to not have hardcoded label names, and use a constant to define it.

Reply:
@otaviof What is the purpose of this label? To identify the backing CRD, it's name and version is sufficient, right?

Reply:
@otaviof On a closer look, I realized that we need label selectors for backing CR also. Something like this:
`MatchLabels  map[string]string `json:"matchLabels"``
I will send a separate PR for the same, right away.

Reply:
I send PR: https://github.com/redhat-developer/service-binding-operator/pull/43
Please review and approve!